05/app.js

```diff
@@ -1,69 +1,39 @@
-const apiUrl = 'http://localhost:3000/users';
+import ApiHandler from './ApiHandler.js';
+const resource = '/users';
 
-const submitBtn = document.querySelector('.form__submit');
+const form = document.querySelector('form');
 document.addEventListener('DOMContentLoaded', init);
 
 function init() {
-	loadUsers();
-	submitBtn.addEventListener('click', getDataFromInput);
-}
-
-function loadUsers() {
-	const promise = fetchGet(apiUrl);
-
-	promise.then(data => insertUsers(data)).catch(err => console.error(err));
-}
-
-function fetchGet(url) {
-	return fetch(url).then(resp => {
-		if (resp.ok) {
-			return resp.json();
+	const apiHandler = new ApiHandler();
+	apiHandler.load(resource).then(resp => insertUsers(resp));
+
+	form.addEventListener('submit', getDataFromInput);
+
+	function getDataFromInput(e) {
+		// i tak strone przeladowuje :|
+		e.preventDefault();
+		e.stopPropagation();
+		const name = document.querySelector('.form__field--first-name').value;
+		const lastName = document.querySelector('.form__field--last-name').value;
+
+		if (name != '' && lastName != '') {
+			const userData = {
+				firstName: name,
+				lastName: lastName,
+			};
+			apiHandler.create(resource, userData).then(data => console.log(data));
```

Comment:
Ponieważ nie przeładowuje to warto ponownie załadować wszystkie wycieczki (lub dodać tą, która teraz została wprowadzona).